### PR TITLE
Move the claims generate csv service

### PR DESCRIPTION
**Pull request by Nitemaeric**
## Context

The `GenerateClaimsCsv` service is scoped to the `Claims::Claim` model but the service isn't scoped to the model.

## Changes proposed in this pull request

- Add the `CSV` acronym and rename the affected classes.
- Move the `Claims::GenerateClaimsCsv` service to `Claims::Claim::GenerateCSV`.
- ~Add an `ApplicationService` for a service object base class in place of the `ServicePattern` concern.~ This change is now at https://github.com/DFE-Digital/itt-mentor-services/pull/984

## Guidance to review

- Tests should all pass

### CODE REVIEW

File: config/initializers/inflections.rb [hunk 1]
```diff
@@ -14,4 +14,5 @@
 ActiveSupport::Inflector.inflections(:en) do |inflect|
   inflect.acronym "DfE"
   inflect.acronym "QA"
+  inflect.acronym "CSV"
```
Comment (Kizr) [inline config/initializers/inflections.rb:17]: ❤️

File: app/services/claims/provider/generate_csv.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 require "csv"
 
-class Claims::Provider::GenerateCsv
+class Claims::Provider::GenerateCSV
```
Comment (Kizr) [inline app/services/claims/provider/generate_csv.rb:3]: Could this inherit from `ApplicationService`?

File: app/services/gias/csv_transformer.rb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 module Gias
-  class CsvTransformer
+  class CSVTransformer
     include ServicePattern
```
Comment (JamieCleare2525) [inline app/services/gias/csv_transformer.rb:3]: This service should now inherit from ApplicationService, and include ServicePattern removed

**Review by ollietreend — COMMENTED** [review]
I also [said this in Slack](https://ukgovernmentdfe.slack.com/archives/C06962474AC/p1725031614516479?thread_ts=1725021191.547129&cid=C06962474AC) but I'm copying it here so it's in the proper place.

This PR looks like it covers two independent sets of changes:

1. Introduce the acronym "CSV"
2. Introduce an ApplicationService base class

@Nitemaeric would you mind tidying this up so it's either two clean PRs, or at least two clean commits, please? 🙏🏻 Right now these changes are a bit mixed.

I'm not against either change 🙂  they both seem reasonable enough.

But it would be good to go all-in when introducing such changes. For example, let's refactor every service object to inherit from ApplicationService rather than leaving a mix of the concern vs inheritance.

**Comment by Nitemaeric** [comment]
I'll refactor. I forgot to mention in the PR that if all is good, I'll update all services. I just did the 1 service to show an example.

File: app/services/claims/provider/generate_csv.rb (hunk 2 above)
Comment (Nitemaeric) [reply app/services/claims/provider/generate_csv.rb:3]: I've moved the `ApplicationService` introduction into https://github.com/DFE-Digital/itt-mentor-services/pull/984.

[the remark this answers is not in the document]
Comment (Nitemaeric) [reply app/services/claims/provider/generate_csv.rb:3]: I've moved the `ApplicationService` introduction into https://github.com/DFE-Digital/itt-mentor-services/pull/984.

File: app/services/gias/csv_downloader.rb [hunk 4]
```diff
@@ -1,5 +1,5 @@
 module Gias
-  class CsvDownloader
+  class CSVDownloader
```
[the remark this answers is not in the document]
Comment (Nitemaeric) [reply app/services/gias/csv_downloader.rb:2]: I've moved the `ApplicationService` introduction into https://github.com/DFE-Digital/itt-mentor-services/pull/984.

File: app/services/gias/csv_importer.rb [hunk 5]
```diff
@@ -1,7 +1,7 @@
 require "csv"
 
 module Gias
-  class CsvImporter
+  class CSVImporter
     include ServicePattern
```
[the remark this answers is not in the document]
Comment (Nitemaeric) [reply app/services/gias/csv_importer.rb:5]: I've moved the `ApplicationService` introduction into https://github.com/DFE-Digital/itt-mentor-services/pull/984.

File: app/services/gias/csv_transformer.rb (hunk 3 above)
Comment (Nitemaeric) [reply app/services/gias/csv_transformer.rb:3]: I've moved the `ApplicationService` introduction into https://github.com/DFE-Digital/itt-mentor-services/pull/984.

**Review by JamieCleare2525 — APPROVED** [review]

**Pull request merged**